grafana-plugin/src/containers/IntegrationContainers/ExpandedIntegrationRouteDisplay/ExpandedIntegrationRouteDisplay.tsx

```diff
@@ -143,41 +189,105 @@ export const ExpandedIntegrationRouteDisplay: React.FC<ExpandedIntegrationRouteD
               ) : (
                 <VerticalGroup spacing="sm">
                   <Text customTag="h6" type="primary">
-                    Use routing template
+                    {hasLabels ? 'Alerts matched by' : 'Use routing template'}
                   </Text>
 
-                  <HorizontalGroup spacing="xs">
-                    <div className={cx('input', 'input--align')}>
-                      <MonacoEditor
-                        value={channelFilterTemplate}
-                        disabled={true}
-                        height={MONACO_INPUT_HEIGHT_SMALL}
-                        data={templates}
-                        showLineNumbers={false}
-                        monacoOptions={MONACO_READONLY_CONFIG}
+                  <RenderConditionally shouldRender={hasLabels}>
+                    <VerticalGroup>
+                      <div className={cx('labels-panel')}>
+                        <RadioButtonGroup
+                          options={QueryBuilderOptions}
+                          value={routingOption}
+                          onChange={async (option) => {
+                            if (option === ROUTING_OPTION.TEMPLATE) {
+                              await alertReceiveChannelStore.saveChannelFilter(channelFilterId, {
+                                filtering_term: channelFilter.filtering_term || '',
+                                filtering_term_type: FilteringTermType.jinja2,
+                              });
+                            }
+
+                            if (option === ROUTING_OPTION.LABELS) {
+                              await alertReceiveChannelStore.saveChannelFilter(channelFilterId, {
+                                filtering_labels: channelFilter.filtering_labels || [],
+                                filtering_term_type: FilteringTermType.labels,
+                              });
+                            }
+
+                            setRoutingOption(option);
+                          }}
```

Comment:
Can we extract this part to the const above return? like `onRoutingOptionChange` or similar